istioctl/cmd/revision.go

```diff
@@ -0,0 +1,659 @@
+// Copyright Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"istio.io/istio/pkg/kube"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+	"istio.io/api/label"
+	"istio.io/api/operator/v1alpha1"
+	"istio.io/istio/operator/cmd/mesh"
+	operator_istio "istio.io/istio/operator/pkg/apis/istio"
+	iopv1alpha1 "istio.io/istio/operator/pkg/apis/istio/v1alpha1"
+	"istio.io/istio/operator/pkg/manifest"
+	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/operator/pkg/util/clog"
+	"istio.io/istio/pkg/config"
+	v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachinery_schema "k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+type revisionArgs struct {
+	manifestsPath string
+	name          string
+	verbose       bool
+	sections      []string
+	output        string
+}
+
+const (
+	IstioOperatorCRSection = "cr"
+	ControlPlaneSection    = "cp"
+	GatewaysSection        = "gw"
+	PodsSection            = "po"
+
+	// TODO: This should be moved to istio/api:label package
+	istioTag = "istio.io/tag"
+)
+
+var (
+	validFormats = []string{"table", "json"}
+
+	defaultSections = []string{
+		IstioOperatorCRSection,
+		ControlPlaneSection,
+		GatewaysSection,
+	}
+
+	verboseSections = []string{
+		IstioOperatorCRSection,
+		ControlPlaneSection,
+		GatewaysSection,
+		PodsSection,
+	}
+)
+
+var (
+	istioOperatorGVR = apimachinery_schema.GroupVersionResource{
+		Group:    iopv1alpha1.SchemeGroupVersion.Group,
+		Version:  iopv1alpha1.SchemeGroupVersion.Version,
+		Resource: "istiooperators",
+	}
+
+	revArgs = revisionArgs{}
+)
+
+func revisionCommand() *cobra.Command {
+	revisionCmd := &cobra.Command{
+		Use:     "revision",
+		Short:   "Revision centric view of Istio deployment",
+		Aliases: []string{"rev"},
+	}
+	revisionCmd.PersistentFlags().StringVarP(&revArgs.manifestsPath, "manifests", "d", "", mesh.ManifestsFlagHelpStr)
+	revisionCmd.PersistentFlags().BoolVarP(&revArgs.verbose, "verbose", "v", false, "print customizations")
```

Comment:
Let's keep this consistent.
```suggestion
	revisionCmd.PersistentFlags().BoolVarP(&revArgs.verbose, "verbose", "v", false, "Enable verbose output.")
```